environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java

```diff
@@ -238,6 +242,13 @@ private void fireContainerInitializedEvent() {
      */
     public Event<Object> event() {
         checkState();
+        if (event == null) {
+            synchronized (this) {
```

Comment:
So this is basically check whether the Event built-in bean has been already initialized/used right? Still not quite sure why it's volatile. Usage within extension could mean execution in different thread?